app/services/messaging/channel.rb

```diff
@@ -9,31 +9,35 @@ def initialize
     @metrics = Metrics.with_object(self)
   end
 
+  delegate :transaction, to: ActiveRecord::Base
+
   attr_reader :metrics
 
+  # The channel implementation is responsible for creating a Communication
+  # and delivery details. This should return the communication object that was created.
```

Comment:
This PR moves the responsibility of recording the communication and the detailable to the messaging channels. There's two reasons:
- Since the messaging channels can return different responses - Twilio returns an object, BSNL will return some JSON, it's hard to work with them at the `NotificationDispatch` layer. The channels should ideally return the same shape of result that also hides the implementation. A `communication` abstracts this very well.
-  Any other consumer of this API in our codebase would also probably need to record communications.

Side effect: We will start recording communications for OTPs. We _could_ override this behaviour in the OTP channel class but there isn't a strong reason to do it. We send ~3k OTPs in a month in IHCI, so it should be fine to record communications and detailables for them.